src/niscope/system_tests/test_system_niscope.py

```diff
@@ -261,18 +261,18 @@ def test_waveform_processing(session):
     session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
     with session.initiate():
         session.add_waveform_processing(niscope.enums.ArrayMeasurement.DERIVATIVE)
-        processed_waveforms = session.channels[test_channels]._fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
+        processed_waveforms = session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
         session.clear_waveform_processing()
-        unprocessed_waveforms = session.channels[test_channels]._fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
+        unprocessed_waveforms = session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
 
     assert len(processed_waveforms) == test_num_channels * test_num_records
     assert len(unprocessed_waveforms) == test_num_channels * test_num_records
     # Here the idea is to leave a large margin to not test too specifically for any returned values but to demonstrate that the waveform processing does
     # undeniably cause a consistent shift in the values returned. The "0" exception for processed is due to the nature of derivatives -- if two samples
     # next to each other are identical, the derivative will be 0.
     for processed, unprocessed in zip(processed_waveforms, unprocessed_waveforms):
-        assert abs(unprocessed) < 1
-        assert abs(processed) > 1 or processed == 0
+        assert abs(unprocessed.result) < 1
+        assert abs(processed.result) > 1 or processed == 0
```

Comment:
`processed.result == 0`

Reply:
Gotcha. Was just pushing so I could run the tests on my testing machine, thanks for the catch.